src/timestamp.rs

```diff
@@ -164,148 +164,187 @@ impl Interval {
             stop: Timestamp(self.stop.0 + duration_ns),
         }
     }
+    // Translate interval by duration_ns on both sides.
+    pub fn translate(self, duration_ns: i64) -> Self {
+        Self {
+            start: Timestamp(self.start.0 + duration_ns),
+            stop: Timestamp(self.stop.0 + duration_ns),
+        }
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    #[test]
-    fn test_s() {
-        assert_eq!(Timestamp::parse("123.4 s"), Ok(Timestamp(123_400_000_000)));
-    }
+    mod timestamp {
```

Comment:
Is there a reason for the nested module? We're already in `timestamp.rs` so I'm not sure I see the need, and it creates an especially large diff.

Reply:
The file has two separate things, `Timestamp` and `Interval`. All the existing tests were for `Timestamp` but the new test is for `Interval` so there needs to be some way to distinguish them. The other option would be to rename all the tests, e.g. `test_timestamp_foo` and `test_interval_bar` but that seemed unpleasant.